src/app/wallet/address-book/address-book.component.ts

```diff
@@ -69,13 +70,24 @@ export class AddressBookComponent implements OnInit {
     */
   addAddressToBook() {
     if (this.label && this.label.trim() && !this.isMine) {
-      this._rpc.call('manageaddressbook', ['newsend', this.address, this.label])
-        .subscribe(
-          response => this.rpc_addAddressToBook_success(response),
-          error => this.rpc_addAddressToBook_failed(error));
+      if (['Locked', 'Unlocked, staking only'].indexOf(this._rpc.state.get('encryptionstatus')) !== -1) {
```

Comment:
We should rather introduce a new state, maybe call it locked, and assign to true or false based on this condition.
Have a look at rpc-state.service - We can subscribe to encryptionstatus by doing
```
state.observe('encryptionstatus')
  .subscribe(
    status => state.set('locked', ['Locked', 'Unlocked, staking only'].includes(status)));
```
That way we can just use `state.get('locked')` for these types of conditions.